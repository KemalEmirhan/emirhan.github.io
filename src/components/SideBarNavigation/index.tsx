import Link from 'next/link';
import React from 'react';

const SideBarNavigation = () => {
  return (
    <aside className='hidden lg:block border-b md:border-b-0 md:border-r border-black border-dashed md:pr-4'>
      <nav className='flex md:flex-col items-end gap-4 py-4 md:py-0 md:pr-4'>
        <Link
          href='/'
          className='text-gray-700 hover:text-gray-900 transition-colors duration-150 underline text-xl uppercase'
        >
          About
        </Link>
        <Link
          href='/socials'
          className='text-gray-700 hover:text-gray-900 transition-colors duration-150 underline text-xl uppercase'
        >
          Socials
        </Link>
        {/* <Link
          href='/bookmarks'
          className='text-gray-700 hover:text-gray-900 transition-colors duration-150 underline text-xl uppercase'
        >
          Bookmarks
        </Link> */}
      </nav>
    </aside>
  );
};

export default SideBarNavigation;
