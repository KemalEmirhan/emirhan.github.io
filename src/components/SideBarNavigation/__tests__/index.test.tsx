import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import SideBarNavigation from '../index';

describe('SideBarNavigation', () => {
  it('should render', () => {
    const { container } = render(<SideBarNavigation />);
    expect(container).toMatchSnapshot();
  });

  it('should render all navigation links', () => {
    const { getByRole } = render(<SideBarNavigation />);

    const aboutLink = getByRole('link', { name: 'About' });
    const socialsLink = getByRole('link', { name: 'Socials' });

    expect(aboutLink).toBeInTheDocument();
    expect(socialsLink).toBeInTheDocument();
  });

  it('should have correct href attributes', () => {
    const { getByRole } = render(<SideBarNavigation />);

    const aboutLink = getByRole('link', { name: 'About' });
    const socialsLink = getByRole('link', { name: 'Socials' });

    expect(aboutLink).toHaveAttribute('href', '/');
    expect(socialsLink).toHaveAttribute('href', '/socials');
  });

  it('should have proper accessibility attributes', () => {
    const { getByRole, getAllByRole } = render(<SideBarNavigation />);

    const nav = getByRole('navigation');
    expect(nav).toBeInTheDocument();

    const links = getAllByRole('link');
    expect(links).toHaveLength(2);
  });
});
