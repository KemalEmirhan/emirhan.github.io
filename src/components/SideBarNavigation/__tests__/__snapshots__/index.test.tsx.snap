// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SideBarNavigation > should render 1`] = `
<div>
  <aside
    class="hidden lg:block border-b md:border-b-0 md:border-r border-black border-dashed md:pr-4"
  >
    <nav
      class="flex md:flex-col items-end gap-4 py-4 md:py-0 md:pr-4"
    >
      <a
        class="text-gray-700 hover:text-gray-900 transition-colors duration-150 underline text-xl uppercase"
        href="/"
      >
        About
      </a>
      <a
        class="text-gray-700 hover:text-gray-900 transition-colors duration-150 underline text-xl uppercase"
        href="/socials"
      >
        Socials
      </a>
    </nav>
  </aside>
</div>
`;
